docs/api_comparison.md

```diff
@@ -0,0 +1,254 @@
+```
+
+```{note}
+`h3-py` has no runtime dependencies on other libraries, so the standard
+`pip install h3` will not install any additional libraries.
+
+However, `h3.api.numpy_int` does require `numpy`.
+To install `numpy` (if it isn't already) along
+with `h3`, you can run `pip install h3[numpy]`.
+```
+
+### `h3.api.memview_int`
+
+H3 indexes are represented as `uint64`s, using Python
+[`memoryview` objects](https://docs.python.org/dev/library/stdtypes.html#memoryview)
+for collections.
+
+This API has the same benefits as `numpy_int`, except it uses
+(the less well-known but dependency-free) `memoryview`.
+In fact, `h3.api.numpy_int` is essentially just a light wrapper around 
+`h3.api.memview_int`.
+
+```python
+>>> import h3.api.memview_int as h3
+>>> h = h3.geo_to_h3(0, 0, 0)
+>>> h
+578536630256664575
+
+>>> mv = h3.hex_ring(h, 1)
+>>> mv
+<MemoryView of 'array' at 0x11188c710>
+
+>>> mv[0]
+578782920861286399
+
+>>> list(mv)
+[578782920861286399,
+ 578044049047420927,
+ 577973680303243263,
+ 578677367745019903,
+ 579169948954263551]
+```
+
+````{warning}
+When using the `memview_int` API with the `numpy` library, note that the
+[`numpy.array`](https://numpy.org/doc/stable/reference/generated/numpy.array.html)
+conversion function **creates a copy** of the data.
+
+On the other hand,
+[`numpy.asarray`](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html)
+**does not create a copy**, and the
+result points to the same memory location as the original `memoryview` object.
+
+For example, consider the setup:
+
+```python
+>>> import h3.api.memview_int as h3
+>>> import numpy as np
+>>> h = h3.geo_to_h3(0, 0, 0)
+>>> mv = h3.hex_ring(h, 1)
+>>> list(mv)
+[578782920861286399,
+ 578044049047420927,
+ 577973680303243263,
+ 578677367745019903,
+ 579169948954263551]
+```
+
+Running `a = np.array(mv)` **creates a copy** of the memory view, so
+modifying `mv` leaves `a` unchanged:
+
+```python
+>>> a = np.array(mv)
+>>> mv[0] = 0
+>>> a
+array([578782920861286399, 578044049047420927, 577973680303243263,
+       578677367745019903, 579169948954263551], dtype=uint64)
+```
+
+Running `a = np.asarray(mv)` **does not create a copy**, so modifying `mv` also
+modifies `a`:
+
+```python
+>>> mv = h3.hex_ring(h, 1)
+>>> a = np.asarray(mv)
+>>> mv[0] = 0
+>>> a
+array([                 0, 578044049047420927, 577973680303243263,
+       578677367745019903, 579169948954263551], dtype=uint64)
+```
+````
+
+
+## API speed comparison
+
+Here's an example task to compare compute times across APIs:
+
+```python
+import h3
+import h3.api.numpy_int
+
+def foo(h3_lib, N=100):
+    h   = h3_lib.geo_to_h3(0, 0, 9)
+    out = h3_lib.k_ring(h, N)
+    out = h3_lib.compact(out)
+    
+    return out
+```
+
+Using the standard API with Python `str` and `set` objects:
+
+```python
+>>> %%timeit
+... out = foo(h3.api.basic_str)
+53.3 ms ± 473 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
+```
+
+Using `h3.api.numpy_int`:
```

Comment:
Maybe worthwhile to include `basic_int` here too?

Reply:
Sure. Adding a notebook here: https://github.com/uber/h3-py-notebooks/pull/7